WooCommerce/Classes/POS/Controllers/PointOfSaleItemsController.swift

```diff
@@ -25,6 +32,7 @@ protocol PointOfSaleItemsControllerProtocol {
     private let paginationTracker: AsyncPaginationTracker
     private var childPaginationTrackers: [POSItem: AsyncPaginationTracker] = [:]
     private let itemProvider: PointOfSaleItemServiceProtocol
+    private var itemType: POSItemType = .products
```

Comment:
Based on a discussion p1742206171389629/1741916795.290219-slack-C070SJRA8DP, we could also consider how `ItemListState` could be used to represent the Products | Coupons data structure. 

Maybe something like this? 🤔 

<img width="715" alt="image" src="https://github.com/user-attachments/assets/b27b59e5-c079-4286-9e93-c77471311d7f" />

The current selection could be tracked on a state or separately.